tools/profiler/public/ProfilerMarkers.h

```diff
@@ -218,6 +225,14 @@ class MOZ_RAII AutoProfilerTextMarker {
       "AutoProfilerTextMarker::~AutoProfilerTextMarker(): name=%s",
       mMarkerName
     );
```

Comment:
This recording assertion needs to be removed too, otherwise the spurious mismatches will remain.